tests/integration/lxd/conftest.py

```diff
@@ -91,13 +93,45 @@ def tmp_instance(
 
 @pytest.fixture(autouse=True, scope="module")
 def installed_lxd():
-    """Ensure lxd is installed, or skip the test if we cannot."""
-    if shutil.which("lxc") is None or shutil.which("lxd") is None:
+    """Ensure lxd is installed, or skip the test if we cannot.
+
+    If the environment has CRAFT_PROVIDERS_TESTS_ENABLE_LXD_INSTALL=1,
+    force the installation of LXD if uninstalled.
+    """
+    if sys.platform != "linux":
+        pytest.skip(f"lxd not supported on {sys.platform}")
+
+    if os.environ.get("CRAFT_PROVIDERS_TESTS_ENABLE_LXD_INSTALL") == "1":
+        lxd.install()
+    elif not lxd.is_installed():
```

Comment:
would this not silently skip tests, should the conditional be the other way around, skip if the env is not set, falback to just installing

Reply:
If lxd is installed, the tests will run.
If lxd is not installed and CRAFT_PROVIDERS_TESTS_ENABLE_LXD_INSTALL == 1, lxd will be installed and the tests will run.
Otherwise, the tests will be skipped.

There is a bug here I fixed for both Multipass and LXD.